misc/shunting_yard.c

```diff
@@ -0,0 +1,225 @@
+/**
+ * @file
+ * @brief [Shunting Yard Algorithm](https://en.wikipedia.org/wiki/Shunting_yard_algorithm)
+ * @details From Wikipedia: In computer science,
+ * the shunting yard algorithm is a method for parsing arithmetical or logical expressions, or a combination of both, specified in infix notation.
+ * It can produce either a postfix notation string, also known as Reverse Polish notation (RPN), or an abstract syntax tree (AST).
+ * The algorithm was invented by Edsger Dijkstra and named the "shunting yard" algorithm because its operation resembles that of a railroad shunting yard.
+ * @author [CascadingCascade](https://github.com/CascadingCascade)
+ */
+
+#include <assert.h>     /// for assertion
+#include <stdio.h>      /// for IO operations
+#include <stdlib.h>     /// for memory management
+#include <string.h>     /// for string operations
+#include <ctype.h>      /// for isdigit()
+
+/**
+ * @brief Helper function that returns each operator's precedence
+ * @param operator the operator to be queried
+ * @returns the operator's precedence
+ */
+int getPrecedence(char operator) {
+    switch (operator) {
+        case '+':
+        case '-': {
+            return 1;
+        }
+        case '*':
+        case '/': {
+            return 2;
+        }
+        case '^': {
+            return 3;
+        }
+        default:{
+            fprintf(stderr,"Error: Invalid operator\n");
+            return -1;
+        }
+    }
+}
+
+/**
+ * @brief Helper function that returns each operator's associativity
+ * @param operator the operator to be queried
+ * @returns '1' if the operator is left associative
+ * @returns '0' if the operator is right associative
+ */
+int getAssociativity(char operator) {
+    switch (operator) {
+        case '^': {
+            return 0;
```

Comment:
Shouldn't you add `break` for these statements?

Reply:
I thought `break` aren't necessary since they `return` instead?